oceannavigator/frontend/src/components/SubsetPanel.jsx

```diff
@@ -54,6 +71,86 @@ function SubsetPanel(props) {
 
     return [lat_min, lat_max, long_min, long_max];
   };
+//   const handleVariableSelection = (selectedVariableIds) => {
+//     setOutputVariables(selectedVariableIds);
+
+
+
+//   // const selectedVariableObjects = subset_variables.filter((v) =>
+//   //   selectedVariableIds.includes(v.id)
+//   // );
+//   // if (!Array.isArray(selectedVariableIds)) {
+//   //   console.warn('selectedVariableIds is not an array:', selectedVariableIds);
+//   //   setShowDepthSelector(false);
+// //     return; // or set to empty array: selectedVariableIds = [];
+// // }
+
+// // Now safe to proceed
+// const selectedVariableObjects = subset_variables.filter((v) =>
+//     selectedVariableIds.includes(v.id)
+// );
+
+//   const anyHasDepth = selectedVariableObjects.some((v) => v.two_dimensional === false);
+//   setShowDepthSelector(anyHasDepth);
+
+//   if (anyHasDepth) {
+//     // Use first 3D variable to fetch depths
+//     const first3DVariable = selectedVariableObjects.find(v => v.two_dimensional === false);
+//     if (first3DVariable) {
+//       const currentDatasetId = props.dataset.id;
+//       GetDepthsPromise(props.dataset.id, first3DVariable.id)
+//         .then((result) => {
+//            if (currentDatasetId === props.dataset.id) {
+//           setAvailableDepths(result.data);
+//           setOutputDepth(result.data[1].id);}
+//         })
+//         .catch((err) => console.error("Failed to fetch depths:", err));
+//     }
+//   } else {
+//     setAvailableDepths([]);
+//     setOutputDepth(null);
+//   }
+// };
+
+const handleVariableSelection = (selectedVariableIds) => {
+  // Handle the -5 case (and any other non-array values)
+  if (!Array.isArray(selectedVariableIds)) {
+    console.log('selectedVariableIds is not an array:', selectedVariableIds, typeof selectedVariableIds);
+    setOutputVariables([]);
+    setShowDepthSelector(false);
+    setAvailableDepths([]);
+    setOutputDepth(props.dataset.depth);
+    return;
+  }
+
+  setOutputVariables(selectedVariableIds);
+
+  const selectedVariableObjects = subset_variables.filter((v) =>
+    selectedVariableIds.includes(v.id)
+  );
+
+  const anyHasDepth = selectedVariableObjects.some((v) => v.two_dimensional === false);
+  setShowDepthSelector(anyHasDepth);
+
+  if (anyHasDepth) {
+    const first3DVariable = selectedVariableObjects.find(v => v.two_dimensional === false);
+    if (first3DVariable) {
+      const currentDatasetId = props.dataset.id;
+      GetDepthsPromise(props.dataset.id, first3DVariable.id)
+        .then((result) => {
+          if (currentDatasetId === props.dataset.id) {
+            setAvailableDepths(result.data);
+            setOutputDepth(result.data[1].id);
+          }
+        })
+        .catch((err) => console.error("Failed to fetch depths:", err));
+    }
+  } else {
+    setAvailableDepths([]);
+    setOutputDepth(props.dataset.depth);
+  }
+};
```

Comment:
You can condense all of this logic into a few lines by filtering for 3d variables on line 129. i.e:

```
    const variablesWithDepth = subset_variables.filter(
      (v) => selectedVariableIds.includes(v.id) && v.two_dimensional === false
    );

    if (variablesWithDepth.length > 0) {
     get depths and set state
    } else {
     set state for no depths
    }
```